textile/objects-features.textile

```diff
@@ -0,0 +1,145 @@
+---
+title: Objects Features
+section: client-lib-development-guide
+index: 65
+jump_to:
+  Help with:
+    - Objects Features Overview#overview
+---
+
+h2(#overview). Overview
+
+This document outlines the feature specification for the Objects feature of the Realtime system. It is currently under development and stored separately from the main specification to simplify the initial implementation of the feature in other SDKs. Once completed, it will be moved to the main "features":../features spec.
+
+Objects feature enables clients to store shared data as "objects" on a channel. When an object is updated, changes are automatically propagated to all subscribed clients in realtime, ensuring each client always sees the latest state.
+
+h3(#realtime-objects). RealtimeObjects
+
+* @(RTO1)@ @Objects#getRoot@ function:
```

Comment:
I have been caught out a few times by this method call hanging because I forgot to attach the channel. Does JS implement any sort of implicit attach?

Reply:
Currently, we only throw error for `ChannelState.detached` and `ChannelState.failed`, I feel we should also check for `ChannelState.initialized` state. I guess, we do the same in case of chat-sdk

Reply:
I see objects as a feature of the core SDK, on the same level as, say, presence. We have presence features that implicitly perform an attach (e.g. `enter`).

Reply:
OK, feel free to resolve this one